tests/test_postgres.py

```diff
@@ -1748,6 +1748,25 @@ def test_has_monitoring_privileges_exception(self, conn_mock, mock_is_superuser)
         # THEN a None value was returned
         assert has_monitoring is None
 
+    @patch("barman.postgres.PostgreSQLConnection.has_backup_privileges")
+    @patch("barman.postgres.PostgreSQLConnection.connect")
+    def test_current_size(self, conn_mock, has_backup_privileges_mock):
```

Comment:
For completeness, you can add test cases for:

* `has_backup_privileges` == `True`
* `PostgresConnectionError` and `psycopg2.Error` being triggered.